### PR TITLE
Bug 2015498: Fixes add capacity for MCG standalone

### PR DESCRIPTION
**Pull request by vbnrh**
and fixes regression with external mode add capacity

MCG standalone with no add capacity
Screenshot from 2021-11-10 15-37-45

Normal storagesystem with add capacity
Screenshot from 2021-11-10 15-39-32

Signed-off-by: Vineet Badrinath <vineetbnath@gmail.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx [hunk 1]
```diff
@@ -128,6 +129,9 @@ export const setActionFlags = (
       break;
     default:
   }
+  isMCG
+    ? flagDispatcher(setFlag(MCG_STANDALONE, true))
+    : flagDispatcher(setFlag(MCG_STANDALONE, false));
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:134]:
```suggestion
   flagDispatcher(setFlag(MCG_STANDALONE, isMCG))
```

File: frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx [hunk 2]
```diff
@@ -18,8 +18,8 @@ export const getGenericActions = () => [...Kebab.factory.common];
 
 export const getActionsForOCS = () => [addStorage, ...getGenericActions()];
 
-export const getActions = (kind: string) => {
-  if (referenceForModel(OCSServiceModel).toLowerCase() === kind.toLowerCase())
+export const getActions = (kind: string, onlyMCG: boolean) => {
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx:21]:
```suggestion
export const getActions = (kind: string, isOnlyMCG: boolean) => {
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx (hunk 1 above)
Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:134]: my brain seems to live on hard mode :)

File: frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx (hunk 2 above)
Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx:21]: done

File: frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx [hunk 3]
```diff
@@ -18,8 +18,8 @@ export const getGenericActions = () => [...Kebab.factory.common];
...
+  if (referenceForModel(OCSServiceModel).toLowerCase() === kind.toLowerCase() && !onlyMCG)
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx:22]: Don't we need to block for `OCS_INDEPENDENT_FLAG` here as well ?

Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/odf-system/actions.tsx:22]: true ! completely forgot about this

**Comment by afreen23** [comment]
> /retest /lgtm image

This should not show add capacity at all.
This is tested wrongly or the fix is missing something . Please recheck @vbnrh @rexagod 
/lgtm cancel

**Comment by vbnrh** [comment]
> > /retest /lgtm image
> 
> This should not show add capacity at all. This is tested wrongly or the fix is missing something . Please recheck @vbnrh @rexagod /lgtm cancel

@rexagod is this mcg only deployment ? If it is not then, add capacity showing up is correct

**Review by afreen23 — APPROVED** [review]
/lgtm

**Comment by rexagod** [comment]
EDIT: Updated the comment: https://github.com/openshift/console/pull/10426#issuecomment-965136915

**Comment by afreen23** [comment]
/hold

Tests are failing

**Pull request merged**